File: src/main/java/red/tetracube/iot/smartigloohub/messagingbroker/payloads/MessagingBrokerClientAuthenticationRequest.java
package red.tetracube.iot.smartigloohub.messagingbroker.payloads;

import com.fasterxml.jackson.annotation.JsonProperty;
import io.smallrye.common.constraint.NotNull;

import javax.validation.constraints.NotEmpty;

public record MessagingBrokerClientAuthenticationRequest(@NotNull @JsonProperty("client_id") String clientId,
                                                         @NotNull @NotEmpty @JsonProperty("username") String username,
                                                         @NotNull @NotEmpty @JsonProperty("password") String password) {
}

File: src/main/kotlin/red/tetracube/iot/smartigloohub/security/api/payloads/MessagingBrokerClientAuthenticationRequest.kt
package red.tetracube.iot.smartigloohub.security.api.payloads

import com.fasterxml.jackson.annotation.JsonCreator
import com.fasterxml.jackson.annotation.JsonProperty
import io.smallrye.common.constraint.NotNull
import javax.validation.constraints.NotEmpty

data class MessagingBrokerClientAuthenticationRequest @JsonCreator constructor(
    @field:NotNull
    @JsonProperty("client_id")
    val clientId: String? = null,

    @field:NotNull
    @field:NotEmpty
    @JsonProperty("username")
    val username: String? = null,

    @field:NotNull
    @field:NotEmpty
    @JsonProperty("password")
    val password: String? = null
)
